pkg/db/db.go

```diff
@@ -0,0 +1,117 @@
+package db
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+)
+
+func getConnectionString(db *config.DB) string {
+	return fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", db.Host, db.Port, db.User, db.Password, db.Name)
+}
+
+type Database struct {
+	tableName string
+	pool      *pgxpool.Pool
+}
```

Comment:
The database structure (or alias) is supposed to represent the database client. Is there a reason to attach a specific table name to it? what will happen when we have more than on table?

Reply:
if we dont store the table we need to pass the table name from the caller function, if you suggest doing it, I can update it.

Reply:
I think you missed the point I was trying to make
```suggestion
type Database struct {
	pool      *pgxpool.Pool
	tables   map[string]string 
}
```

or even better, (as out client only work on specific tables and is not designed to be generic):
```suggestion
type Database struct {
	pool      *pgxpool.Pool
	tables   struct {
	    providers: string
	}
}
```